test/runtests.jl

```diff
@@ -61,49 +63,56 @@ end
 end
 
 @testset "Plots" begin
-    @testset "data as $dtype" for dtype in [:Vectors, :Matrices]
+    @testset "data as $dtype" for dtype in [:Vectors, :Matrices, Symbol("Vectors of vectors")]
         if dtype == :Vectors
             x, y, z = randn(10), randn(10), randn(10)
-        else
+        elseif dtype == :Matrices
             x, y, z = randn(10,2), randn(10,2), randn(10,2)
+        else
+            x, y, z = [rand(10), rand(20)], [rand(10), rand(20)], [rand(10), rand(20)]
         end
 
+
         @testset "One array" begin
             @test plot(x*m)                    isa Plots.Plot
             @test plot(x*m, ylabel="x")        isa Plots.Plot
             @test plot(x*m, ylims=(-1,1))      isa Plots.Plot
             @test plot(x*m, ylims=(-1,1) .* m) isa Plots.Plot
-            @test plot(x*m, yunit=u"km")       isa Plots.Plot
+            dtype ≠ Symbol("Vectors of vectors") && @test plot(x*m, yunit=u"km") isa Plots.Plot
+            @test plot(x -> x^2, x*m)          isa Plots.Plot
         end
 
         @testset "Two arrays" begin
             @test plot(x*m, y*s)                    isa Plots.Plot
             @test plot(x*m, y*s, xlabel="x")        isa Plots.Plot
             @test plot(x*m, y*s, xlims=(-1,1))      isa Plots.Plot
             @test plot(x*m, y*s, xlims=(-1,1) .* m) isa Plots.Plot
-            @test plot(x*m, y*s, xunit=u"km")       isa Plots.Plot
+            dtype ≠ Symbol("Vectors of vectors") && @test plot(x*m, y*s, xunit=u"km") isa Plots.Plot
```

Comment:
```suggestion
            if dtype == Symbol("Vectors of vectors") 
                @test_broken plot(x*m, y*s, xunit=u"km") isa Plots.Plot
            else
                @test plot(x*m, y*s, xunit=u"km") isa Plots.Plot
            end
```
This clarifies the intention.